packages/gestalt/src/contexts/I18nProvider.js

```diff
@@ -0,0 +1,40 @@
+// @flow strict
+import { type Context, createContext, useContext } from 'react';
+
+/**
+ * To add new translations:
+ * - Create a type for the component's translations
+ * - Add component translation type to I18nContextType keyed by component name
+ * - Add component translation type to useI18nContext return type
+ * - When consuming useI18nContext in component, throw an error if any translation isn't available
+ */
+
+type TextTranslations = {|
+  accessibilityHidePasswordLabel: string,
+  accessibilityShowPasswordLabel: string,
+|};
+
+type I18nContextType = ?{|
+  Text: TextTranslations,
```

Comment:
Just a thought. A single type might be easier to scan if this keeps growing.  Is there any particular reason to break down in cmp types rather than inlining?

Reply:
Can you describe your idea for inlining?

Reply:
suggestion - you might be able to do something like this so you can refine valid inputs/outputs for this:
```
type TextTranslations = {|
  Text: {| 
      accessibilityHidePasswordLabel: string,
  |},
  Sheet: {|
      somethingSomethingLabel: string
   |}
|}

type ValidComponent = $Keys<TextTranslations>
...
function useI18nContext<C: ValidComponent>(componentName: C): $ElementType<TextTranslations, C> 
```

i _feel_ like that should work but don't quote me on that 😅

Reply:
nm, figured it out!